docker-java/src/test/java/com/github/dockerjava/cmd/CmdIT.java

```diff
@@ -37,6 +42,32 @@ public DockerCmdExecFactory createExecFactory() {
             public DockerCmdExecFactory createExecFactory() {
                 return new OkHttpDockerCmdExecFactory().withConnectTimeout(30 * 1000);
             }
+        },
+        HTTPCLIENT5(true) {
+            @Override
+            public DockerCmdExecFactory createExecFactory() {
+                class FakeFactory extends DelegatingDockerCmdExecFactory implements DockerClientConfigAware {
+
+                    private DefaultDockerCmdExecFactory dockerCmdExecFactory;
+
+                    @Override
+                    public final DockerCmdExecFactory getDockerCmdExecFactory() {
+                        return dockerCmdExecFactory;
+                    }
+
+                    @Override
+                    public void init(DockerClientConfig dockerClientConfig) {
+                        dockerCmdExecFactory = new DefaultDockerCmdExecFactory(
+                            new ApacheDockerHttpClient.Factory()
+                                .dockerClientConfig(dockerClientConfig)
+                                .build(),
+                            dockerClientConfig.getObjectMapper()
+                        );
+                        dockerCmdExecFactory.init(dockerClientConfig);
+                    }
+                }
+                return new FakeFactory();
```

Comment:
Not a big deal but... why is there no standard factory class for this transport? I'm a bit unclear as to why there would be `NettyDockerCmdExecFactory`, `JerseyDockerCmdExecFactory` and `OkHttpDockerCmdExecFactory` but no corresponding one for this client.

Reply:
okhttp's and jersey's `DockerCmdExecFactory` are deprecated (see #1351) in favour of their `DockerHttpClient` versions. I have not migrated Netty yet, but plan to.

Eventually, `DockerCmdExecFactory` will become an internal class of `core`, and all commands will share their `exec`s

Since this PR adds a new module, there is no point in adding a deprecated type.